### DISCUSSION TITLE
PCB

### DISCUSSION THREAD

**Discussion by krisc-informatica** [krisc-informatica/arduino-smart-bike-trainer#4] [Q&A]
I have designed an accompanying PCB for the project, although I'm no electronics specialist. The design can be found at https://oshwlab.com/kris.cardinaels/arduino-smart-bike-trainer.

Several extensions will be added to this board, a.o.:

- adding LCD (1602A) connector to display data to the user
- adding general button connectors for interaction

Of course, depending on the availability of Arduino pins. Suggestions are welcome.

**Comment by frankhommers** [comment]
If you are ordering your PCB, count me in for two or three ;-)

**Reply by krisc-informatica** [reply]
If you have experience with designing PCBs, feel free to check and comment. Will update for the extra's I mentioned.

**Reply by frankhommers** [reply]
Unfortunately I don't ;-)

**Comment by StFalagar** [comment]
Hi.... I am also interested in this... I have the Tacx T1901...  Not sure if you have seen this post... seems they have it working already...

http://www.timetriallingforum.co.uk/index.php?/topic/131123-old-tacx-flow-into-smart-trainer/

I am also a developer although have no experience with arduino, pc boards etc. etc. but would love to get the trainer working on Zwift etc.

**Reply by krisc-informatica** [reply]
Hello,

I saw that one also and was inspired from it.

The current status of my project is that the speed and rpm work correctly in both FTMS and CSC profiles (separate projects). For the actual control of the brake I have to do more tests... and find a dual/tripple channel oscilloscope. Pulses are generated and are long enough but I'm not sure about the timing.

**Reply by StFalagar** [reply]
Sounds great.... 

Will be keeping an eye out here...   I wish there was something more I could do to contribute though...

**Reply by frankhommers** [reply]
@krisc-informatica I do have an oscilloscope. And also a T1682. Can I help out?

**Reply by StFalagar** [reply]
Have you guys seen this repo?  https://github.com/totalreverse/ttyT1941

**Reply by krisc-informatica** [reply]
Hello, this one in particular I have not seen, but there are others that use the motor brake. That one uses TTY communication which is not the same as the pwm-signal I need for the eddy current brake.

**Reply by krisc-informatica** [reply]
> @krisc-informatica I do have an oscilloscope. And also a T1682. Can I help out?

Could you make images of the brake in action with the synchronization signal and the pwm signal together on the oscilloscope? Sync is the green cable (23VAC) and PWM is the red cable in the connector. The pwm signal is only generated when you effectively make some speed. 
If you could make those images of the signals for the different grade values on the steer mount, would be great.

Thanks in advance,
Kris

**Reply by StFalagar** [reply]
I think my T1901 also uses the Eddy Current Brake but there's another guy on that repo that is also trying to make it work with a T1901... he said using the cable he was able to get readings into Fortius-Ant on Windows and is now concentrating on moving it onto the RPi:

Quote:
I've made a cable to connect the brake to the USB/TTL converter (FT232RL based). I've connected this to a Windows PC running @WouterJD's FortiusANT and have confirmed the USB is operational and the connector is receiving signals from the brake. I will revert to the RPi (instead of PC) and use the code you suggest in your wiki

https://github.com/totalreverse/ttyT1941/issues/21

**Reply by frankhommers** [reply]
@krisc-informatica I added you on Skype. Maybe we can check stuff out during the call when I take the steps for scoping the thing you need. My skype name is the same as my GitHub name.
